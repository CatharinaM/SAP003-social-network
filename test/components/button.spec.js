// importamos la funcion que vamos a testear
import Button from '../../src/components/button.js';

describe('Button', () => {
  it('should have the correct structure', () => {
    expect(Button({ title: 'Text' })).toMatchSnapshot();
  });

  it('should call handleClick on button is clicked', () => {
    const onClick = jest.fn();
    const button = Button({ title: 'Text', onClick });
    button.click();

    expect(onClick.mock.calls).toHaveLength(1);
  });
});
