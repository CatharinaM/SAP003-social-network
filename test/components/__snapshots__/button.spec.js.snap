// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have the correct structure 1`] = `
<button
  class="primary-button"
>
  Text
</button>
`;
